timed/permissions.py

```diff
@@ -141,3 +168,150 @@ class IsNotBilledAndVerfied(BasePermission):
 
     def has_object_permission(self, request, view, obj):
         return not obj.billed or obj.verified_by_id is None
+
+
+class IsInternal(IsAuthenticated):
+    """Allows access only to internal employees."""
+
+    def has_permission(self, request, view):
+        if not super().has_permission(request, view):  # pragma: no cover
+            return False
+
+        if request.method not in SAFE_METHODS:
+            try:
+                employment = employment_models.Employment.objects.get_at(
+                    user=request.user, date=date.today()
+                )
+                return not employment.is_external
+            except employment_models.Employment.DoesNotExist:
+                raise PermissionDenied("User has no employment")
+
+        return True
+
+    def has_object_permission(self, request, view, obj):
+        if not super().has_object_permission(request, view, obj):  # pragma: no cover
+            return False
+
+        employment = employment_models.Employment.objects.get_at(
+            user=request.user, date=date.today()
+        )
+        if employment:
+            return not employment.is_external
+        return False  # pragma: no cover
+
+
+class IsExternal(IsAuthenticated):
+    """Allows access only to external employees."""
+
+    def has_permission(self, request, view):
+        if not super().has_permission(request, view):  # pragma: no cover
+            return False
+
+        if request.method not in SAFE_METHODS:
+            try:
+                employment = employment_models.Employment.objects.get_at(
+                    user=request.user, date=date.today()
+                )
+            except employment_models.Employment.DoesNotExist:  # pragma: no cover
+                raise PermissionDenied("User has no employment")
+
+            if employment:
+                return employment.is_external
+        return True
+
+    def has_object_permission(self, request, view, obj):
+        if not super().has_object_permission(request, view, obj):  # pragma: no cover
+            return False
+
+        employment = employment_models.Employment.objects.get_at(
+            user=request.user, date=date.today()
+        )
+        if employment:
+            return employment.is_external
+        return False  # pragma: no cover
+
+
+class IsManager(IsAuthenticated):
+    """Allows access only to assignees with manager role."""
+
+    def has_permission(self, request, view):
+        if not super().has_permission(request, view):  # pragma: no cover
+            return False
+
+        if request.method not in SAFE_METHODS:
```

Comment:
No method check here either.

Also, you should `return` boolean in every case, which doesn't happen in this method.